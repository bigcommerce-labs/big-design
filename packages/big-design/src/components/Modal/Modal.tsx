import { CloseIcon } from '@bigcommerce/big-design-icons';
import React from 'react';
import { createPortal } from 'react-dom';

import { uniqueId } from '../../utils';
import { Button } from '../Button';
import { H2, HeadingProps } from '../Typography';

import {
  StyledModal,
  StyledModalActions,
  StyledModalBody,
  StyledModalClose,
  StyledModalContent,
  StyledModalHeader,
} from './styled';

export interface ModalProps {
  backdrop: boolean;
  isOpen: boolean;
  closeOnClickOutside: boolean;
  closeOnEscKey: boolean;
  variant: 'modal' | 'dialog';
  onClose(): void;
}

interface ModalState {
  initialBodyOverflowY: string;
  modalContainer: HTMLDivElement | null;
}

export interface ModalActionsProps {
  withBorder?: boolean;
}

export interface ModalHeaderProps {
  withBorder?: boolean;
}

const ModalActions: React.FC<ModalActionsProps> = ({ children, withBorder }) => (
  <StyledModalActions justifyContent="flex-end" withBorder={withBorder}>
    {children}
  </StyledModalActions>
);

const ModalHeader: React.FC<ModalHeaderProps> = ({ children, withBorder }) => {
  return (
    <StyledModalHeader withBorder={withBorder}>
      {typeof children === 'string' ? <H2 margin="none">{children}</H2> : children}
    </StyledModalHeader>
  );
};

export class Modal extends React.PureComponent<ModalProps, ModalState> {
  static defaultProps: Partial<ModalProps> = {
    backdrop: true,
    isOpen: false,
    closeOnClickOutside: false,
    closeOnEscKey: true,
    onClose: () => null,
    variant: 'modal',
  };

  static Actions = ModalActions;
  static Body = StyledModalBody;
  static Header = ModalHeader;

  readonly state: ModalState = {
    initialBodyOverflowY: '',
    modalContainer: null,
  };

  private modalRef = React.createRef<HTMLDivElement>();
  private readonly headerUniqueId = uniqueId('modal_header_');

  componentDidMount() {
    const modalContainer = document.createElement('div');

    document.body.appendChild(modalContainer);
    this.setState({ modalContainer });
  }

  componentWillUnmount() {
    const { initialBodyOverflowY, modalContainer } = this.state;

    if (modalContainer) {
      document.body.removeChild(modalContainer);
    }

    document.body.style.overflowY = initialBodyOverflowY;
  }

  componentDidUpdate(prevProps: ModalProps) {
    // Check that the previous state was not open and is now open before auto focusing on modal
    if (!prevProps.isOpen && this.props.isOpen) {
      this.autoFocus();
      this.setState({ initialBodyOverflowY: document.body.style.overflowY || '' });
    }

    this.props.isOpen
      ? (document.body.style.overflowY = 'hidden')
      : (document.body.style.overflowY = this.state.initialBodyOverflowY);
  }

  render() {
    const { backdrop, isOpen, variant } = this.props;
    const { modalContainer } = this.state;

    const modalContent = (
      <StyledModal
        onKeyDown={this.onKeyDown}
        onClick={this.onClickAway}
        ref={this.modalRef}
        backdrop={backdrop}
        variant={variant}
      >
        <StyledModalContent variant={variant} aria-labelledby={this.headerUniqueId} flexDirection="column">
          {this.renderClose()}
          {this.renderChildren()}
        </StyledModalContent>
      </StyledModal>
    );

    return isOpen && modalContainer ? createPortal(modalContent, modalContainer) : null;
  }

  private renderClose() {
    const { onClose, variant } = this.props;

    return (
      variant === 'modal' && (
        <StyledModalClose>
          <Button onClick={onClose} iconOnly={<CloseIcon title="Close" />} variant="subtle" />
        </StyledModalClose>
      )
    );
  }

  private renderChildren() {
    const { children } = this.props;

    return React.Children.map(children, child => {
      if (React.isValidElement(child) && child.type === Modal.Header) {
        return React.cloneElement(child as React.ReactElement<HeadingProps>, {
          id: this.headerUniqueId,
        });
      }

      return child;
    });
  }

  private autoFocus = () => {
    if (this.props.isOpen && this.modalRef.current) {
      this.modalRef.current.focus();
    }
  };

  private onClickAway = (event: React.MouseEvent) => {
    if (this.props.closeOnClickOutside && this.modalRef.current === event.target) {
      this.props.onClose();
    }
  };

  private onKeyDown = (event: React.KeyboardEvent) => {
    if (this.props.closeOnEscKey && event.key === 'Escape') {
      this.props.onClose();
    }
  };
}
