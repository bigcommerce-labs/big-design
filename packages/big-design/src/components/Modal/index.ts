export { Modal, type ModalProps, type ModalAction } from './Modal';
