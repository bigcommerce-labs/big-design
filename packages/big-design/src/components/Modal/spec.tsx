import 'jest-styled-components';
import React from 'react';
import { fireEvent, render } from 'react-testing-library';

import { Modal } from './Modal';

test('render open modal', () => {
  const text = 'This is a modal';
  const { queryByText } = render(<Modal isOpen={true}>{text}</Modal>);

  expect(document.body).toMatchSnapshot();
  expect(queryByText(text)).toBeInTheDocument();
});

test('render open modal without backdrop', () => {
  const text = 'This is a modal';
  const { queryByText } = render(
    <Modal isOpen={true} backdrop={false}>
      {text}
    </Modal>,
  );

  expect(document.body).toMatchSnapshot();
  expect(queryByText(text)).toBeInTheDocument();
});

test('render closed modal', () => {
  const text = 'This is a modal';
  const { queryByText } = render(<Modal isOpen={false}>{text}</Modal>);

  expect(document.body).toMatchSnapshot();
  expect(queryByText(text)).not.toBeInTheDocument();
});

test('open/hides when props changes', () => {
  const text = 'This is a modal';
  const { queryByText, rerender } = render(<Modal isOpen={false}>{text}</Modal>);

  expect(queryByText(text)).not.toBeInTheDocument();
  rerender(<Modal isOpen={true}>{text}</Modal>);
  expect(queryByText(text)).toBeInTheDocument();
});

test('triggers onClose when pressing esc', () => {
  const text = 'This is a modal';
  const onClose = jest.fn();

  const { queryByText } = render(
    <Modal isOpen={true} onClose={onClose}>
      {text}
    </Modal>,
  );

  const element = queryByText(text) as HTMLElement;

  fireEvent.keyDown(element, { key: 'Escape' });

  expect(onClose).toHaveBeenCalled();
});

test('does not trigger onClose when pressing esc and closeOnEscKey is false', () => {
  const text = 'This is a modal';
  const onClose = jest.fn();

  const { queryByText } = render(
    <Modal isOpen={true} onClose={onClose} closeOnEscKey={false}>
      {text}
    </Modal>,
  );

  const element = queryByText(text) as HTMLElement;

  fireEvent.keyDown(element, { key: 'Escape' });

  expect(onClose).not.toHaveBeenCalled();
});

test('trigger onClose when clicking outside the modal', () => {
  const text = 'This is a modal';
  const onClose = jest.fn();

  const { queryByRole } = render(
    <Modal isOpen={true} onClose={onClose} closeOnClickOutside={true}>
      {text}
    </Modal>,
  );

  const element = queryByRole('dialog') as HTMLElement;

  fireEvent.click(element);

  expect(onClose).toHaveBeenCalled();
});

test('do not trigger onClose when clicking outside the modal and closeOnClickOutside is false', () => {
  const text = 'This is a modal';
  const onClose = jest.fn();

  const { queryByRole } = render(
    <Modal isOpen={true} onClose={onClose}>
      {text}
    </Modal>,
  );

  const element = queryByRole('dialog') as HTMLElement;

  fireEvent.click(element);

  expect(onClose).not.toHaveBeenCalled();
});

test('do not trigger onClose when clicking inside the modal', () => {
  const onClose = jest.fn();

  const { queryByTestId } = render(
    <Modal isOpen={true} onClose={onClose}>
      <p data-testid="inside-modal">Content</p>
    </Modal>,
  );

  const element = queryByTestId('inside-modal') as HTMLElement;

  fireEvent.click(element);

  expect(onClose).not.toHaveBeenCalled();
});

test('render close button on modal variation', () => {
  const text = 'This is a modal';
  const onClose = jest.fn();

  const { queryByTitle } = render(
    <Modal isOpen={true} onClose={onClose} variant="modal">
      {text}
    </Modal>,
  );

  expect(queryByTitle('Close')).toBeInTheDocument();
});

test('do not render close button on dialog variation', () => {
  const text = 'This is a modal';
  const onClose = jest.fn();

  const { queryByTitle } = render(
    <Modal isOpen={true} onClose={onClose} variant="dialog">
      {text}
    </Modal>,
  );

  expect(queryByTitle('Close')).not.toBeInTheDocument();
});

test('do not pull focus to open modal that is rerendered', () => {
  const text = 'This is a modal';
  const { queryByText, queryByRole, rerender } = render(
    <Modal isOpen={false}>
      {text}
      <input id="focusTest" />
    </Modal>,
  );

  // Modal not opened
  expect(queryByText(text)).not.toBeInTheDocument();

  // Modal Opened
  rerender(
    <Modal isOpen={true}>
      {text}
      <input id="focusTest" />
    </Modal>,
  );

  // Expect Modal to have focus
  expect(queryByText(text)).toBeInTheDocument();
  expect(document.activeElement).toBe(queryByRole('dialog'));

  const input = document.getElementById('focusTest');

  // Focus on input
  if (input !== null) {
    input.focus();
    expect(input).toHaveFocus();

    // Rerender as open
    rerender(
      <Modal isOpen={true}>
        {text}
        <input id="focusTest" />
      </Modal>,
    );

    // Expect input to still have focus and not modal
    expect(input).toHaveFocus();
    expect(document.activeElement).not.toBe(queryByRole('dialog'));
  }
});
