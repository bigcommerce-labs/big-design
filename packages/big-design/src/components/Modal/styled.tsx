import { rgba } from 'polished';
import styled, { css } from 'styled-components';

import { defaultTheme, remCalc } from '../../theme';
import { Flex } from '../Flex';

import { ModalProps } from './Modal';

export const StyledModal = styled.div.attrs({
  'aria-modal': true,
  role: 'dialog',
  tabIndex: -1,
})<Partial<ModalProps>>`
  height: 100%;
  left: 0;
  position: fixed;
  top: 0;
  width: 100%;

  ${props =>
    props.backdrop &&
    props.variant &&
    css`
      background: ${() => rgba(props.theme.colors.secondary70, props.variant === 'dialog' ? 0.5 : 0.7)};
    `}
`;

export const StyledModalContent = styled(Flex)<{ variant: ModalProps['variant'] }>`
  background: ${({ theme }) => theme.colors.white};
  box-sizing: border-box;
  position: fixed;

  ${({ variant }) =>
    variant === 'dialog' &&
    css`
      ${({ theme }) => theme.elevation.floating};

      max-width: ${({ theme }) => theme.breakpointValues.tablet};
      left: 50%;
      top: 50%;
      transform: translate(-50%, -50%);
      width: 90%;
    `}

  ${({ variant }) =>
    variant === 'modal' &&
    css`
      height: 100%;
      width: 100%;

      ${({ theme }) => theme.breakpoints.tablet} {
        ${({ theme }) => theme.elevation.floating};

        height: auto;
        left: 50%;
        max-height: ${remCalc(664)};
        max-width: ${({ theme }) => theme.breakpointValues.tablet};
        top: 50%;
        transform: translate(-50%, -50%);
      }
    `}
`;

export const StyledModalActions = styled(Flex)``;

export const StyledModalClose = styled.div`
  position: absolute;
  top: ${({ theme }) => theme.spacing.xxSmall};
  right: ${({ theme }) => theme.spacing.xxSmall};

  ${({ theme }) => theme.breakpoints.tablet} {
    display: none;
  }
`;

export const StyledModalBody = styled.div`
  flex-grow: 1;
  overflow-y: auto;
`;

StyledModal.defaultProps = { theme: defaultTheme };
StyledModalContent.defaultProps = { theme: defaultTheme };
StyledModalActions.defaultProps = { theme: defaultTheme };
StyledModalClose.defaultProps = { theme: defaultTheme };
