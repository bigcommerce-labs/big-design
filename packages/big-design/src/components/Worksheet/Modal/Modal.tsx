import React, { useCallback, useEffect, useMemo, useState } from 'react';

import { typedMemo } from '../../../utils';
import { Modal } from '../../Modal';
import { useEditableCell, useStore } from '../hooks';
import { WorksheetItem, WorksheetModalColumn } from '../types';

interface WorksheetModalProps<Item> {
  column: WorksheetModalColumn<Item>;
}

const InternalWorksheetModal = <T extends WorksheetItem>({ column }: WorksheetModalProps<T>) => {
  const { config, hash } = column;
  const { header, render } = config;

  const isModalOpen = useStore(useMemo(() => (state) => state.openedModal === hash, [hash]));
  const selectedCell = useStore(useMemo(() => (state) => state.selectedCells[0], []));
  const setOpenModal = useStore((state) => state.setOpenModal);

  const { handleChange } = useEditableCell<T>(selectedCell);

  const [newValue, setNewValue] = useState<unknown>(null);

  useEffect(() => {
    if (selectedCell) {
      setNewValue(selectedCell.value);
    }
  }, [selectedCell]);

  const handleSave = useCallback(() => {
    if (selectedCell && newValue !== null && newValue !== selectedCell.value) {
      handleChange(newValue);
    }

    setOpenModal(null);
  }, [handleChange, newValue, selectedCell, setOpenModal]);

  const renderedContent = useMemo(() => {
    const onChange = (newValue: unknown) => {
      if (newValue !== undefined) {
        setNewValue(newValue);
      }
    };

    return selectedCell ? render(selectedCell.value, onChange) : null;
  }, [selectedCell, render]);

  return (
    <Modal
      actions={[
        { text: 'Cancel', variant: 'subtle', onClick: () => setOpenModal(null) },
        {
          text: 'Save',
          onClick: handleSave,
        },
      ]}
      closeOnClickOutside={false}
      closeOnEscKey={true}
      header={header}
      isOpen={isModalOpen}
      onClose={() => setOpenModal(null)}
    >
      {renderedContent}
    </Modal>
  );
};

export const WorksheetModal = typedMemo(InternalWorksheetModal);
