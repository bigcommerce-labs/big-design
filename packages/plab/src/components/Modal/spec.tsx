import 'jest-styled-components';
import React from 'react';
import { fireEvent, render } from 'react-testing-library';

import { Modal } from './Modal';

test('render open modal', () => {
  const text = 'This is a modal';
  const { queryByText } = render(<Modal isOpen={true}>{text}</Modal>);

  expect(document.body).toMatchSnapshot();
  expect(queryByText(text)).toBeInTheDocument();
});

test('render open modal without backdrop', () => {
  const text = 'This is a modal';
  const { queryByText } = render(
    <Modal isOpen={true} backdrop={false}>
      {text}
    </Modal>,
  );

  expect(document.body).toMatchSnapshot();
  expect(queryByText(text)).toBeInTheDocument();
});

test('render closed modal', () => {
  const text = 'This is a modal';
  const { queryByText } = render(<Modal isOpen={false}>{text}</Modal>);

  expect(document.body).toMatchSnapshot();
  expect(queryByText(text)).not.toBeInTheDocument();
});

test('open/hides when props changes', () => {
  const text = 'This is a modal';
  const { queryByText, rerender } = render(<Modal isOpen={false}>{text}</Modal>);

  expect(queryByText(text)).not.toBeInTheDocument();
  rerender(<Modal isOpen={true}>{text}</Modal>);
  expect(queryByText(text)).toBeInTheDocument();
});

test('triggers onRequestClose when pressing esc', () => {
  const text = 'This is a modal';
  const onRequestClose = jest.fn();

  const { queryByText } = render(
    <Modal isOpen={true} onRequestClose={onRequestClose}>
      {text}
    </Modal>,
  );

  const element = queryByText(text) as HTMLElement;

  fireEvent.keyDown(element, { key: 'Escape' });

  expect(onRequestClose).toHaveBeenCalled();
});

test('does not trigger onRequestClose when pressing esc and requestCloseOnEscKey is false', () => {
  const text = 'This is a modal';
  const onRequestClose = jest.fn();

  const { queryByText } = render(
    <Modal isOpen={true} onRequestClose={onRequestClose} requestCloseOnEscKey={false}>
      {text}
    </Modal>,
  );

  const element = queryByText(text) as HTMLElement;

  fireEvent.keyDown(element, { key: 'Escape' });

  expect(onRequestClose).not.toHaveBeenCalled();
});

test('trigger onRequestClose when clicking outside the modal', () => {
  const text = 'This is a modal';
  const onRequestClose = jest.fn();

  const { queryByRole } = render(
    <Modal isOpen={true} onRequestClose={onRequestClose}>
      {text}
    </Modal>,
  );

  const element = queryByRole('dialog') as HTMLElement;

  fireEvent.click(element);

  expect(onRequestClose).toHaveBeenCalled();
});

test('do not trigger onRequestClose when clicking outside the modal and requestCloseOnClickOutside is false', () => {
  const text = 'This is a modal';
  const onRequestClose = jest.fn();

  const { queryByRole } = render(
    <Modal isOpen={true} onRequestClose={onRequestClose} requestCloseOnClickOutside={false}>
      {text}
    </Modal>,
  );

  const element = queryByRole('dialog') as HTMLElement;

  fireEvent.click(element);

  expect(onRequestClose).not.toHaveBeenCalled();
});

test('do not trigger onRequestClose when clicking inside the modal', () => {
  const onRequestClose = jest.fn();

  const { queryByTestId } = render(
    <Modal isOpen={true} onRequestClose={onRequestClose}>
      <p data-testid="inside-modal">Content</p>
    </Modal>,
  );

  const element = queryByTestId('inside-modal') as HTMLElement;

  fireEvent.click(element);

  expect(onRequestClose).not.toHaveBeenCalled();
});
