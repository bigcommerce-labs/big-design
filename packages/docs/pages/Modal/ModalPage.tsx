import { Button, H0, H1, H2, Link, Modal, Text } from '@bigcommerce/big-design';
import React from 'react';

import { Code, CodePreview } from '../../components';
import { ModalPropTable } from '../../PropTables';

export default () => (
  <>
    <H0>Modal</H0>

    <Text>
      A modal appears as a layer on top of the primary interface. Modals disrupt users from interacting with the page
      until they complete a specific task.{' '}
      <Link href="https://design.bigcommerce.com/components/modals" target="_blank">
        Modals Design Guidelines
      </Link>
      .
    </Text>

    <CodePreview>
      {/* jsx-to-string:start */}
      {function Example() {
        const [isOpen, setIsOpen] = React.useState(false);

        return (
          <>
            <Button onClick={() => setIsOpen(true)}>Open Modal</Button>

            <Modal
              actions={[
                { text: 'Cancel', onClick: () => setIsOpen(false) },
                { text: 'Apply', onClick: () => setIsOpen(false) },
              ]}
              header="Modal Title"
              isOpen={isOpen}
              onClose={() => setIsOpen(false)}
              closeOnEscKey={true}
              closeOnClickOutside={false}
            >
              <Text>
                Ea tempor sunt amet labore proident dolor proident commodo in exercitation ea nulla sunt pariatur. Nulla
                sunt ipsum do eu consectetur exercitation occaecat labore aliqua. Aute elit occaecat esse ea fugiat
                esse. Reprehenderit sunt ea ea mollit commodo tempor amet fugiat.
              </Text>
              <Text>
                Esse ipsum est consectetur nulla aute deserunt. Anim sint nisi consequat officia adipisicing irure.
                Nulla ea reprehenderit elit eu nostrud sunt veniam dolore ex occaecat qui. Commodo ullamco ut sint dolor
                quis cillum in et enim culpa esse exercitation ad. Eiusmod adipisicing nisi culpa esse laborum cupidatat
                ad pariatur proident. Consectetur ex sint ullamco non ex.
              </Text>
            </Modal>
          </>
        );
      }}
      {/* jsx-to-string:end */}
    </CodePreview>

    <H1>Dialog</H1>

    <Text>
      Setting the variant prop to <Code primary>dialog</Code> results in a simplified version of a Modal. The purpose of
      a dialog is to act as a safety net for a user attempting a destructive action.{' '}
      <Link href="https://design.bigcommerce.com/components/modals-confirmation" target="_blank">
        Modalsâ€“Confirmation Design Guidelines
      </Link>
      .
    </Text>

    <CodePreview>
      {/* jsx-to-string:start */}
      {function Example() {
        const [isOpen, setIsOpen] = React.useState(false);

        return (
          <>
            <Button onClick={() => setIsOpen(true)}>Open Dialog</Button>

            <Modal
              actions={[
                { text: 'Cancel', onClick: () => setIsOpen(false) },
                { text: 'Apply', onClick: () => setIsOpen(false) },
              ]}
              header="Dialog Title"
              isOpen={isOpen}
              onClose={() => setIsOpen(false)}
              closeOnEscKey={true}
              closeOnClickOutside={false}
              variant="dialog"
            >
              <Text>
                Ea tempor sunt amet labore proident dolor proident commodo in exercitation ea nulla sunt pariatur.
              </Text>
            </Modal>
          </>
        );
      }}
      {/* jsx-to-string:end */}
    </CodePreview>

    <H1>API</H1>

    <H2>Modal</H2>

    <ModalPropTable />
  </>
);
