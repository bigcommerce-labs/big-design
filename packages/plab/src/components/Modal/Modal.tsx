import React from 'react';
import { createPortal } from 'react-dom';

import { StyledModal, StyledModalActions, StyledModalContent } from './styled';

export interface ModalProps {
  backdrop: boolean;
  isOpen: boolean;
  requestCloseOnClickOutside: boolean;
  requestCloseOnEscKey: boolean;
  onRequestClose(): void;
}

export class Modal extends React.PureComponent<ModalProps> {
  static defaultProps: Partial<ModalProps> = {
    backdrop: true,
    isOpen: false,
    requestCloseOnClickOutside: true,
    requestCloseOnEscKey: true,
    onRequestClose: () => null,
  };

  static Actions = StyledModalActions;

  private modalRef = React.createRef<HTMLDivElement>();
  private modalContainer = document.createElement('div');

  componentDidMount() {
    document.body.appendChild(this.modalContainer);
  }

  componentWillUnmount() {
    document.body.removeChild(this.modalContainer);
  }

  componentDidUpdate() {
    this.autoFocus();
  }

  render() {
    const { backdrop, children, isOpen } = this.props;

    const modalContent = (
      <StyledModal onKeyDown={this.onKeyDown} onClick={this.onClickAway} ref={this.modalRef} backdrop={backdrop}>
        <StyledModalContent>{children}</StyledModalContent>
      </StyledModal>
    );

    return isOpen && createPortal(modalContent, this.modalContainer);
  }

  private autoFocus = () => {
    if (this.props.isOpen && this.modalRef.current) {
      this.modalRef.current.focus();
    }
  };

  private onClickAway = (event: React.MouseEvent) => {
    if (this.props.requestCloseOnClickOutside && this.modalRef.current === event.target) {
      this.props.onRequestClose();
    }
  };

  private onKeyDown = (event: React.KeyboardEvent) => {
    if (this.props.requestCloseOnEscKey && event.key === 'Escape') {
      this.props.onRequestClose();
    }
  };
}
