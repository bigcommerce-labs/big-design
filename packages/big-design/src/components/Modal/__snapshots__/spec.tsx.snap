// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render closed modal 1`] = `
<body>
  <div />
  <div />
</body>
`;

exports[`render open modal 1`] = `
.c7 {
  vertical-align: middle;
  height: 1.5rem;
  width: 1.5rem;
}

.c3 {
  box-sizing: border-box;
}

.c2 {
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid #D9DCE9;
  border-radius: 0.25rem;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  font-size: 1rem;
  font-weight: 400;
  height: 2.25rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 2rem;
  outline: none;
  padding: 0 1rem;
  pointer-events: auto;
  position: relative;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  white-space: nowrap;
  width: 100%;
  padding: 0 0.5rem;
  background-color: transparent;
  border-color: transparent;
  color: #3C64F4;
}

.c5:focus {
  outline: none;
}

.c5[disabled] {
  border-color: #D9DCE9;
  pointer-events: none;
}

.c5 + .bd-button {
  margin-top: 0.5rem;
}

.c5:active {
  background-color: #DBE3FE;
}

.c5:focus {
  box-shadow: 0 0 0 0.25rem #DBE3FE;
}

.c5:hover:not(:active) {
  background-color: #F0F3FF;
}

.c5[disabled] {
  border-color: transparent;
  color: #D9DCE9;
}

.c6 {
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: inline-grid;
  grid-auto-flow: column;
  grid-gap: 0.5rem;
  visibility: visible;
}

.c0 {
  height: 100%;
  left: 0;
  position: fixed;
  top: 0;
  width: 100%;
  background: rgba(49,52,64,0.7);
}

.c1 {
  background: #FFFFFF;
  box-sizing: border-box;
  position: fixed;
  height: 100%;
  width: 100%;
}

.c4 {
  position: absolute;
  top: 0.25rem;
  right: 0.25rem;
}

@media (min-width:0px) {
  .c3 {
    padding: 1rem;
  }
}

@media (min-width:720px) {
  .c3 {
    padding: 2rem;
  }
}

@media (min-width:720px) {
  .c5 + .bd-button {
    margin-top: 0;
    margin-left: 0.5rem;
  }
}

@media (min-width:720px) {
  .c5 {
    width: auto;
  }
}

@media (min-width:720px) {
  .c1 {
    border-radius: 0.25rem;
    box-shadow: 0px 2px 12px rgba(49,52,64,0.2);
    height: auto;
    left: 50%;
    max-height: 41.5rem;
    max-width: 720px;
    top: 50%;
    -webkit-transform: translate(-50%,-50%);
    -ms-transform: translate(-50%,-50%);
    transform: translate(-50%,-50%);
  }
}

@media (min-width:720px) {
  .c4 {
    display: none;
  }
}

<body>
  <div />
  <div>
    <div
      aria-modal="true"
      class="c0"
      role="dialog"
      tabindex="-1"
    >
      <div
        aria-labelledby="modal_header_0"
        class="c1 c2 c3"
      >
        <div
          class="c4"
        >
          <button
            class="bd-button c5"
            role="button"
            tabindex="0"
          >
            <span
              class="c6"
            >
              <svg
                class="c7"
                fill="currentColor"
                height="24"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="24"
              >
                <title>
                  Close
                </title>
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M18.3 5.71a.996.996 0 00-1.41 0L12 10.59 7.11 5.7A.996.996 0 105.7 7.11L10.59 12 5.7 16.89a.996.996 0 101.41 1.41L12 13.41l4.89 4.89a.996.996 0 101.41-1.41L13.41 12l4.89-4.89c.38-.38.38-1.02 0-1.4z"
                />
              </svg>
            </span>
          </button>
        </div>
        This is a modal
      </div>
    </div>
  </div>
</body>
`;

exports[`render open modal without backdrop 1`] = `
<body>
  <div />
  .c7 {
  vertical-align: middle;
  height: 1.5rem;
  width: 1.5rem;
}

.c3 {
  box-sizing: border-box;
}

.c2 {
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid #D9DCE9;
  border-radius: 0.25rem;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  font-size: 1rem;
  font-weight: 400;
  height: 2.25rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 2rem;
  outline: none;
  padding: 0 1rem;
  pointer-events: auto;
  position: relative;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  white-space: nowrap;
  width: 100%;
  padding: 0 0.5rem;
  background-color: transparent;
  border-color: transparent;
  color: #3C64F4;
}

.c5:focus {
  outline: none;
}

.c5[disabled] {
  border-color: #D9DCE9;
  pointer-events: none;
}

.c5 + .bd-button {
  margin-top: 0.5rem;
}

.c5:active {
  background-color: #DBE3FE;
}

.c5:focus {
  box-shadow: 0 0 0 0.25rem #DBE3FE;
}

.c5:hover:not(:active) {
  background-color: #F0F3FF;
}

.c5[disabled] {
  border-color: transparent;
  color: #D9DCE9;
}

.c6 {
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: inline-grid;
  grid-auto-flow: column;
  grid-gap: 0.5rem;
  visibility: visible;
}

.c0 {
  height: 100%;
  left: 0;
  position: fixed;
  top: 0;
  width: 100%;
}

.c1 {
  background: #FFFFFF;
  box-sizing: border-box;
  position: fixed;
  height: 100%;
  width: 100%;
}

.c4 {
  position: absolute;
  top: 0.25rem;
  right: 0.25rem;
}

@media (min-width:0px) {
  .c3 {
    padding: 1rem;
  }
}

@media (min-width:720px) {
  .c3 {
    padding: 2rem;
  }
}

@media (min-width:720px) {
  .c5 + .bd-button {
    margin-top: 0;
    margin-left: 0.5rem;
  }
}

@media (min-width:720px) {
  .c5 {
    width: auto;
  }
}

@media (min-width:720px) {
  .c1 {
    border-radius: 0.25rem;
    box-shadow: 0px 2px 12px rgba(49,52,64,0.2);
    height: auto;
    left: 50%;
    max-height: 41.5rem;
    max-width: 720px;
    top: 50%;
    -webkit-transform: translate(-50%,-50%);
    -ms-transform: translate(-50%,-50%);
    transform: translate(-50%,-50%);
  }
}

@media (min-width:720px) {
  .c4 {
    display: none;
  }
}

<div>
    <div
      aria-modal="true"
      class="c0"
      role="dialog"
      tabindex="-1"
    >
      <div
        aria-labelledby="modal_header_0"
        class="c1 c2 c3"
      >
        <div
          class="c4"
        >
          <button
            class="bd-button c5"
            role="button"
            tabindex="0"
          >
            <span
              class="c6"
            >
              <svg
                class="c7"
                fill="currentColor"
                height="24"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="24"
              >
                <title>
                  Close
                </title>
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M18.3 5.71a.996.996 0 00-1.41 0L12 10.59 7.11 5.7A.996.996 0 105.7 7.11L10.59 12 5.7 16.89a.996.996 0 101.41 1.41L12 13.41l4.89 4.89a.996.996 0 101.41-1.41L13.41 12l4.89-4.89c.38-.38.38-1.02 0-1.4z"
                />
              </svg>
            </span>
          </button>
        </div>
        This is a modal
      </div>
    </div>
  </div>
</body>
`;
